packages/preview-middleware/src/base/flex.ts

```diff
@@ -90,6 +92,10 @@ export function writeChange(
         }
         const filePath = join(path, fileName + '.' + fileType);
         writeFileSync(filePath, JSON.stringify(data, null, 2));
+        const telemetryData = { category: 'Save', changeType: (data as any).changeType, layer: (data as any).layer };
```

Comment:
1. We don't need to build telemetry data if reporter isn't passed.
2. It's better to extend the parameter type definition on line 80, than using `any`.

Reply:
ok. Extended parameters.